backend/src/api/game/challenge.rs

```diff
@@ -0,0 +1,62 @@
+use actix_web::{post, web, HttpResponse, delete};
+use hive_lib::game_type::GameType;
+use serde::{Deserialize, Serialize};
+
+use crate::server_error::ServerError;
+use crate::model::challenge::GameChallenge;
+use crate::{db::util::DbPool, extractors::auth::AuthenticatedUser};
+
+#[derive(Deserialize)]
+pub struct NewGameRequest {
+    // Whether this challenge should be listed publicly
+    pub public: bool,
+
+    // Whether the game will be ranked
+    pub ranked: bool,
+
+    // Whether the game follows the "tournament" rules, i.e. the queen
+    // cannot be played first. Always true for now
+    pub tournament_queen_rule: bool,
+
+    pub game_type: GameType,
+}
+
+#[derive(Serialize)]
+pub struct NewGameResponse {
+    challenge_url: String,
+}
+
+#[post("/game/challenge")]
+pub async fn create_game_challenge(
+    game: web::Json<NewGameRequest>,
+    auth_user: AuthenticatedUser,
+    pool: web::Data<DbPool>,
+) -> Result<HttpResponse, ServerError> {
+    let challenge = GameChallenge::create(&auth_user.uid, &game, &pool).await?;
+    let challenge_url = format!("/game/challenge/{}", challenge.id);
```

Comment:
okay, answered my own question here... my concern still holds :slightly_smiling_face: 
Maybe we can use https://docs.rs/nanoid/latest/nanoid/

Reply:
turns out psql can generate random uuids for us! i'll switch to those, which solves the discoverability problem and also moves us away from the serial type.